src/secp256k1.c

```diff
@@ -325,6 +326,34 @@ int secp256k1_ec_pubkey_cmp(const secp256k1_context* ctx, const secp256k1_pubkey
     return secp256k1_memcmp_var(out[0], out[1], sizeof(out[0]));
 }
 
+static int secp256k1_ec_pubkey_sort_cmp(const void* pk1, const void* pk2, void *ctx) {
+    return secp256k1_ec_pubkey_cmp((secp256k1_context *)ctx,
+                                     *(secp256k1_pubkey **)pk1,
+                                     *(secp256k1_pubkey **)pk2);
```

Comment:
In #1519, I initially created a `secp256k1_silentpayments_recipient_sort_cmp` function using `(secp256k1_silentpayments_recipient *)r->scan_pubkey` instead of `*(secp256k1_silentpayments_recipient **)r->scan_pubkey`. Everything compiled fine, but the end result was my array was not sorted correctly. Took me awhile to figure out that it was due to using `(typedef *)` instead of `*(typedef **)`.

This could just be a me problem in not having a strong C background, but wanted to mention it in case its worth documenting.

Reply:
Hm, this is documented. The hsort doc says "comparison function is called with two arguments that point to the objects being compared" (which was copied from the qsort_r manpage). If the array you give to hsort consists of elements of type `secp256k1_silentpayments_recipient *`, the comparison function is called with pointers to the elements (`secp256k1_silentpayments_recipient **`).

Reply:
I guess seeing stuff like `*(secp256k1_pubkey **)pk1` can be confusing even to seasoned C programmers at first glance, due to the asterisk being used for two related but different things here (first as a dereference operator and then in the cast for indicating a pointer type). But yeah, I don't think we can do anything to improve this.